static/js/context-menu.js

```diff
@@ -933,39 +950,180 @@ var TreeSwitcherMenu = new (class TreeSwitcherMenu extends ContextMenuBase {
   }
 
   setupMenu() {
-    const lists = [];
-
-    for (const column of TREE_LIST) {
-      const list = document.createElement("ul");
-
-      const menuItems = [];
-      for (const group of column) {
-
-        menuItems.push({
-          html: `<b>${group.name}</b>`,
-          section: "group-" + group.name,
+    const columns = [];
+    const columnBoxes = [];
+
+    for (const groups of TREE_LIST) {
+      const columnBox = document.createElement("div");
+      const column = [];
+      for (const group of groups) {
+        const menuItems = [];
+        const list = document.createElement("ul");
+
+        const label = document.createElement("label");
+        label.classList.add("context-menu-group-label");
+        label.textContent = group.name;
+        column.push({
+          label,
         });
+        columnBox.append(label);
 
         for (const item of group.items) {
           const label = item.label ? item.label : item.value;
           const tree = item.value;
 
-          menuItems.push({
+          const li = this.createMenuItem({
             html: label,
             classNames: ["indent"],
             href: document.location.pathname.replace(/^\/[^\/]+\//, `/${tree}/`)
               + document.location.search
               + document.location.hash,
-            section: "group-" + group.name,
+            attrs: {
+              "data-tree": tree,
+            },
+            useKeys: true,
+          });
+
+          list.append(li);
+          column.push({
+            link: li.firstChild,
           });
         }
+        columnBox.append(list);
       }
+      columns.push(column);
+      columnBoxes.push(columnBox);
+    }
+
+    this.columns = columns;
 
-      this.populateMenu(list, menuItems);
+    this.menu.replaceChildren(...columnBoxes);
+  }
+
+  getCurrentTree() {
+    const m = document.location.pathname.match(/^\/([^\/]+)\//);
+    if (m) {
+      return m[1];
+    }
 
-      lists.push(list);
+    // Fallback
+    return "mozilla-central";
+  }
+
+  focusCurrentTree() {
+    const tree = this.getCurrentTree();
+    const item = this.menu.querySelector(`a[data-tree="${tree}"`);
```

Comment:
I think this selector is missing a closing-bracket, although it seems like our implementation doesn't care and this works fine either way.  (That said, the [CSS grammar does want the closing square brace](https://drafts.csswg.org/selectors-4/#typedef-attribute-selector).)

Reply:
Thanks! filed https://bugzilla.mozilla.org/show_bug.cgi?id=1909096. will fix it shortly